go/client/ctclient/ctclient.go

```diff
@@ -64,8 +66,27 @@ func addChain(ctx context.Context, logClient *client.LogClient) {
 			chain = append(chain, ct.ASN1Cert{Data: block.Bytes})
 		}
 	}
+	if len(chain) == 0 {
+		log.Fatalf("No certificates found in %s", *certChain)
+	}
+
+	// Examine the leaf to see if it looks like a pre-certificate.
+	isPrecert := false
+	leaf, err := x509.ParseCertificate(chain[0].Data)
+	if err == nil {
+		count, _ := x509util.OidInExtensions(x509.OidExtensionCTPoison, leaf.Extensions)
+		if count > 0 {
+			isPrecert = true
+			fmt.Print("Uploading pre-certificate to log\n")
```

Comment:
At the moment all of the normal output from `ctclient` is just `fmt.Printf` -- all of the `log`s are for error conditions.